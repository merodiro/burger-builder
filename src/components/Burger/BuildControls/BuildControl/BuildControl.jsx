import React from 'react'
import PropTypes from 'prop-types'

import classes from './BuildControl.module.css'

const BuildControl = ({ label, added, removed, disabled }) => {
  return (
    <div className={classes.BuildControl}>
      <div className={classes.Label}>{label}</div>
      <button className={classes.Less} onClick={removed} disabled={disabled}>
        Less
      </button>
      <button className={classes.More} onClick={added}>
        More
      </button>
    </div>
  )
}

BuildControl.propTypes = {
  label: PropTypes.oneOf(['Salad', 'Bacon', 'Cheese', 'Meat']).isRequired,
  added: PropTypes.func,
  removed: PropTypes.func,
  disabled: PropTypes.bool.isRequired,
}

export default BuildControl
