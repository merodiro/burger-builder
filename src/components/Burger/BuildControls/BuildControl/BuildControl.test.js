import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import BuildControl from './BuildControl'

describe('<BuildControl>', () => {
  it('renders the same output', () => {
    const component = shallow(<BuildControl label="Bacon" disabled />)
    expect(component).toMatchSnapshot()
  })
})
