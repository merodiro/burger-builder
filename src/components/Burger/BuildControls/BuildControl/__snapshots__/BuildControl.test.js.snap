// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildControl> renders the same output 1`] = `
<div
  className="BuildControl"
>
  <div
    className="Label"
  >
    Bacon
  </div>
  <button
    className="Less"
    disabled="true"
  >
    Less
  </button>
  <button
    className="More"
  >
    More
  </button>
</div>
`;
